bins/stackablectl/src/cmds/stacklets.rs

```diff
@@ -72,50 +76,146 @@ async fn list_cmd(args: &StackletListArgs, common_args: &Cli) -> Result<String,
 
+            // The error table displays optional errors in a structured manner.
+            let mut error_table = Table::new();
+            error_table
+                .set_header(vec!["#", "MESSAGES"])
+                .set_content_arrangement(ContentArrangement::Dynamic)
+                .load_preset(NOTHING);
+
+            let mut product_index = 0;
             for (product_name, products) in stacklets {
                 for product in products {
-                    let endpoints = product
-                        .endpoints
-                        .iter()
-                        .map(|(name, url)| format!("{name}:{url}"))
-                        .collect::<Vec<String>>()
-                        .join("\n");
-
-                    let additional_information = product
-                        .additional_information
-                        .iter()
-                        .map(|(key, value)| format!("{key}:{value}"))
-                        .collect::<Vec<String>>()
-                        .join("\n");
+                    let (conditions, errors) =
+                        process_conditions(product.conditions, &mut product_index, use_color);
 
                     table.add_row(vec![
                         product_name.clone(),
                         product.name,
                         product.namespace.unwrap_or_default(),
-                        endpoints,
-                        additional_information,
+                        conditions,
                     ]);
+
+                    for error in errors {
+                        error_table.add_row(vec![product_index.to_string(), error]);
+                    }
                 }
             }
 
-            Ok(table.to_string())
+            // Only output the error table if there are errors to report.
+            // Currently this is a little awkward, but an upstream PR will make
+            // this more straight forward.
+            Ok(format!(
+                "{}{}",
+                table,
+                if error_table.row_iter().cloned().count() > 0 {
+                    format!("\n\n{}", error_table)
+                } else {
+                    "".into()
+                }
+            ))
         }
-        OutputType::Json => todo!(),
-        OutputType::Yaml => todo!(),
+        OutputType::Json => serde_json::to_string(&stacklets).context(JsonOutputFormatSnafu {}),
+        OutputType::Yaml => serde_yaml::to_string(&stacklets).context(YamlOutputFormatSnafu {}),
+    }
+}
+
+/// Processes conditions for a single stacklet / product. It returns a
+/// concatenated string of conditions (which are colored green and red) to
+/// display next to each listed stacklet in the table. Additionally, it also
+/// returns a list of errors to be displayed underneath the stacklet table.
+fn process_conditions(
+    product_conditions: Vec<DisplayCondition>,
+    product_index: &mut usize,
+    use_color: bool,
+) -> (String, Vec<String>) {
```

Comment:
Might be more clear to return a struct here?

Reply:
Agreed, will update.

Reply:
Changed in 6b1d4ff